x-pack/solutions/security/plugins/security_solution/public/common/components/visualization_actions/visualization_embeddable.tsx

```diff
@@ -36,7 +38,15 @@ const VisualizationEmbeddableComponent: React.FC<VisualizationEmbeddableProps> =
       inputId,
       queryId: id,
     });
-  const { indicesExist } = useSourcererDataView(lensProps.scopeId);
+
+  let { indicesExist } = useSourcererDataView(lensProps.scopeId);
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  const { dataView } = useDataView(lensProps.scopeId);
+
+  if (newDataViewPickerEnabled) {
+    indicesExist = !!dataView?.matchedIndices?.length;
```

Comment:
Using `Boolean()` should improve readability

```suggestion
    indicesExist = Boolean(dataView?.matchedIndices?.length);
```